backend/dataall/modules/shares_base/services/share_notification_service.py

```diff
@@ -175,3 +214,40 @@ def _create_notification_task(self, subject, msg):
                     log.info(f'Notification type : {share_notification_config_type} is not active')
         else:
             log.info('Notifications are not active')
+
```

Comment:
I think we need to review the design a bit because this pattern is new to data.all:
1. an scheduled ECS task gets triggered, reads some metadata from RDS, executes some light business logic and queues Worker tasks
2. the Worker Lambda sends the emails

At first sight it seems like we can simplify the workflow and run everything from ECS

Reply:
I think the only benefit of sending task to queue for worker lambda is if the worker lambda has code bundled in its image that is not there for the ECS container. However this is not the case and both images should have all the backend code at their disposal for whatever data.all task

While separating out all of our backend code into a more microservice aligned design could be useful - it is not in scope for this PR and requires much more design consideration and careful thought

TLDR - I think let's just keep it all in ECS :)

Reply:
I actually had discussed the design with Noah before implementing it and we considered both options:

Option 1: Create a new ECS task that finds pending shares, creates tasks for them, and queues them up, then uses the existing worker Lambda to send emails.

Option 2: Use the send_email_task directly, as you suggested.

I chose Option 1 because I thought having a queue would be beneficial for managing a high volume of outgoing emails. It helps to queue up tasks and avoid potential clashes.

I am open to hearing what others have to say too.

Reply:
I am going to vote for option 2: If we were implementing this from scratch, I would go for a queue + Lambda to process the emails. The part that I do not like form the current architecture is that we are using an ECS task to do very minor things. The new pattern that I would like to see is: scheduled Lambda ---> SQS queue ---> Worker Lambda. But that is out of scope for this PR. We can approve the current scheduled ECS ---> SQS queue ---> Worker Lambda and think about scheduled ECS tasks that could be handled in Lambda in a separate PR

Reply:
+1 after some more thought I think we maybe should just keep all compute for email reminders in ECS for now

I think it should only require a small change to L238 of `share_notification_service` to not use `Worker.queue()` and instead directly call 
```
SESEmailNotificationService.send_email_task(
     subject, message, recipient_groups_list, recipient_email_list
 )
```

I think it could be worthwhile to keep Task creation nonetheless for auditability

Reply:
@noah-paige Added 
```
SESEmailNotificationService.send_email_task(
     subject, message, recipient_groups_list, recipient_email_list
 )
```
directly to `share_notification_service`

Reply:
typo: I voted for option 1, but I am not strong opinionated.